### DISCUSSION TITLE
How can we run superset-frontend on different host, other than localhost?

### DISCUSSION THREAD

**Discussion by hyouinKyouma** [apache/superset#12711] [Q&A / Help]
So I am trying to run superset in dev mode on cloud for which I need to run superset-frontend via ```npm run dev-server``` but this starts the server on localhost, however on cloud I would like to specify the host and port for superset-frontend,

I tried editing the webpack.config.js and webpack.proxy.config.js but the server still starts on localhost.
What can be done in this case to pass the new host for superset-frontend?

**Comment by hyouinKyouma** [comment]
Turns out if we edit the webpack.config.js file and add an additional parameter for for host by passing devserverHost = '0.0.0.0' and then replacing the 'localhost' with ${devserverHost} solves the problem, additionally this also has to be done for the webpack.proxy.config.js file

**Comment by ktmud** [comment]
It is not recommended to run the dev-server in public cloud. You can always run the backend on the cloud, and use `npm run dev-server -- --superset=https://superset-dev.example.com` to proxy a local dev-server to use the public backend.

**Reply by hyouinKyouma** [reply]
Thanks @ktmud this was really helpful, but is there a way to run the backend itself in dev mode too?
